src/Domain.LinnApps/Parts/PartService.cs

```diff
@@ -1,18 +1,46 @@
 ﻿namespace Linn.Stores.Domain.LinnApps.Parts
 {
-    using System.Collections;
+    using System;
     using System.Collections.Generic;
+    using System.Linq;
 
     using Linn.Common.Authorisation;
+    using Linn.Common.Persistence;
     using Linn.Stores.Domain.LinnApps.Exceptions;
+    using Linn.Stores.Domain.LinnApps.ExternalServices;
 
     public class PartService : IPartService
     {
         private readonly IAuthorisationService authService;
 
-        public PartService(IAuthorisationService authService)
+        private readonly IQueryRepository<Supplier> supplierRepository;
+
+        private readonly IRepository<QcControl, int> qcControlRepository;
+
+        private readonly IRepository<Part, int> partRepository;
+
+        private readonly IRepository<PartTemplate, string> templateRepository;
+
+        private readonly IPartPack partPack;
+
+        private readonly ITransactionManager transactionManager;
```

Comment:
Should a lot of this be in a facade service rather than being in the domain?

Reply:
I'd always put as much as possible in the domain but the transaction manager should maybe be the level above. Was this to do with the post-commit triggers?

Reply:
I'm happy with everything being in the domain except that transaction manager. To be honest it might not even be necessary. My thinking was that all I've done in this file is update the PartTemplate object that corresponds to the table. Changes to that object won't be persisted until a Commit() is called, no? A Commit() might already be happening further downstream that catches this anyway so I'll double-check. Anyway, I should definitely move it out of the domain if I do need to keep it in.

Reply:
I think I am getting Commit() confused with SaveChanges()